src/lex/replace.rs

```diff
@@ -312,3 +325,29 @@ fn replace_function(
         .chain(replacements.into_iter().map(|t| Ok(location.with(t))))
         .collect()
 }
+
+fn stringify(args: Vec<Token>) -> Token {
+    let ret: String = args
+        .into_iter()
+        .map(|arg| {
+            match arg {
+                Token::Whitespace(_) => String::from(" "), // Single space in replacement
+                Token::Literal(Literal::Str(s)) => {
+                    // TODO can we unwrap safely?
+                    let new_s = std::str::from_utf8(&s)
+                        .unwrap()
```

Comment:
Hmm, I thought this would fail on `"\xff"` but it seems to work reasonably well:
```c
$ cargo run -- -E
#define str(a) # a
str("\xff") 

 "�" 
```
I'm not sure why though.

Reply:
Without the whitespace, it crashes as expected:
```c
#define str(a) #a 
str("\xff") 
The application panicked (crashed).
Message:  called `Result::unwrap()` on an `Err` value: Utf8Error { valid_up_to: 0, error_len: Some(1) }
Location: src/lex/replace.rs:337
```

Reply:
How do you suggest I handle this? I'd like to avoid writing my own string escape if possible.

Reply:
This would also be helped by https://github.com/jyn514/saltwater/issues/384 I think - instead of greedily turning escapes into the relevant binary, we could instead stringify before processing escape sequences. I think without that stringifying doesn't make much sense anyway since you don't have a clear way to represent binary as text.

Reply:
So should I add this to the `stringify_string` test and we'll handle it later?

Reply:
Yeah, that sounds reasonable.